crates/goose-cli/src/recipe.rs

```diff
@@ -68,3 +79,26 @@ pub fn load_recipe<P: AsRef<Path>>(path: P, log: bool) -> Result<Recipe> {
 
     Ok(recipe)
 }
+
+fn render_content_with_params(
+    content: &str,
+    path: &Path,
+    params: &[(String, String)],
+) -> Result<String> {
+    // Turn params into HashMap
+    let param_map: HashMap<String, String> = params.iter().cloned().collect();
+
+    // Create a Tera context
+    let mut context: TeraContext = TeraContext::new();
+    for (key, value) in &param_map {
+        context.insert(key, value);
+    }
```

Comment:
don't think we need to create the hashmap if we're just directly inserting it into context immediately, you can just iterate over the params slice directly